File: sdk/fsharp/src/HiTechCloudSDK/ProxmoxBackup.fs
namespace HiTechCloud.SDK

/// Proxmox Backup: PBS connection, credentials, usage, snapshots, groups, password, token
type ProxmoxBackupResource(httpClient: HttpClient) =

    /// GET /api/service/{id}/pbs - Get backup connection info
    member this.GetConnectionInfo() =
        async {
            return! httpClient.getAsync("/api/service/{id}/pbs")
        }

    /// GET /api/service/{id}/pbs/credentials - Get backup credentials
    member this.GetCredentials() =
        async {
            return! httpClient.getAsync("/api/service/{id}/pbs/credentials")
        }

    /// GET /api/service/{id}/pbs/usage - Get backup usage
    member this.GetUsage() =
        async {
            return! httpClient.getAsync("/api/service/{id}/pbs/usage")
        }

    /// GET /api/service/{id}/pbs/metrics - Get backup metrics
    member this.GetMetrics() =
        async {
            return! httpClient.getAsync("/api/service/{id}/pbs/metrics")
        }

    /// GET /api/service/{id}/pbs/snapshots - List snapshots
    member this.ListSnapshots() =
        async {
            return! httpClient.getAsync("/api/service/{id}/pbs/snapshots")
        }

    /// GET /api/service/{id}/pbs/groups - List backup groups
    member this.ListGroups() =
        async {
            return! httpClient.getAsync("/api/service/{id}/pbs/groups")
        }

    /// POST /api/service/{id}/pbs/password - Change backup password
    member this.ChangePassword() =
        async {
            return! httpClient.postAsync("/api/service/{id}/pbs/password")
        }

    /// POST /api/service/{id}/pbs/token - Rotate API token
    member this.RotateToken() =
        async {
            return! httpClient.postAsync("/api/service/{id}/pbs/token")
        }

    /// DELETE /api/service/{id}/pbs/token - Revoke API token
    member this.RevokeToken() =
        async {
            return! httpClient.deleteAsync("/api/service/{id}/pbs/token")
        }

File: sdk/clojure/src/hitechcloud/sdk/proxmox_backup.clj
(ns hitechcloud.sdk.proxmox_backup
  (:require [hitechcloud.sdk.http-client :as http]))

;; Proxmox Backup: PBS connection, credentials, usage, snapshots, groups, password, token

(defn get_connection_info "GET /api/service/{id}/pbs - Get backup connection info" [client & [params]] (http-get client "/api/service/{id}/pbs" params))

(defn get_credentials "GET /api/service/{id}/pbs/credentials - Get backup credentials" [client & [params]] (http-get client "/api/service/{id}/pbs/credentials" params))

(defn get_usage "GET /api/service/{id}/pbs/usage - Get backup usage" [client & [params]] (http-get client "/api/service/{id}/pbs/usage" params))

(defn get_metrics "GET /api/service/{id}/pbs/metrics - Get backup metrics" [client & [params]] (http-get client "/api/service/{id}/pbs/metrics" params))

(defn list_snapshots "GET /api/service/{id}/pbs/snapshots - List snapshots" [client & [params]] (http-get client "/api/service/{id}/pbs/snapshots" params))

(defn list_groups "GET /api/service/{id}/pbs/groups - List backup groups" [client & [params]] (http-get client "/api/service/{id}/pbs/groups" params))

(defn change_password "POST /api/service/{id}/pbs/password - Change backup password" [client & [params]] (http-post client "/api/service/{id}/pbs/password" params))

(defn rotate_token "POST /api/service/{id}/pbs/token - Rotate API token" [client & [params]] (http-post client "/api/service/{id}/pbs/token" params))

(defn revoke_token "DELETE /api/service/{id}/pbs/token - Revoke API token" [client & [params]] (http-delete client "/api/service/{id}/pbs/token" params))

